subprojects/platform-jvm/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainIntegrationTest.groovy

```diff
@@ -176,12 +178,11 @@ class JavaToolchainIntegrationTest extends AbstractIntegrationSpec implements Ja
         """
 
         when:
-        executer.withArgument("--no-configuration-cache")
-            .expectDocumentedDeprecationWarning "Requesting JVM vendor IBM_SEMERU. " +
+        executer.expectDocumentedDeprecationWarning "Requesting JVM vendor IBM_SEMERU. " +
             "This behavior has been deprecated. This behavior is scheduled to be removed in Gradle 9.0. " +
             "Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_8.html#ibm_semeru_should_not_be_used"
 
         then:
-        succeeds 'build'
+        succeeds ':build'
```

Comment:
:question: does it still fail running `:help` instead?

Reply:
> :question: does it still fail running `:help` instead?

Running `:help` does not show up the warning that is tested, so I guess the toolchain validation still happens without any source code?

The strange thing is that the toolchain provisioning error does not happen without CC, but the warning yes.

Could be that the eager realization of the toolchain stops before the toolchain is probed?

Reply:
Ok. I assume that the toolchain validation happens even when there is no source code.